okapi-core/src/main/java/org/folio/okapi/util/ModuleUtil.java

```diff
@@ -83,23 +83,28 @@ private static boolean interfaceCheck(
       InterfaceDescriptor[] interfaces, String interfaceStr, String scope) {
     if (interfaceStr == null) {
       return true;
-    } else {
-      if (interfaces != null) {
-        for (InterfaceDescriptor pi : interfaces) {
-          String[] interfaceList = interfaceStr.split(",");
-          for (String interfacePair : interfaceList) {
-            String[] kv = interfacePair.split("=");
-            List<String> gotScope = pi.getScopeArray();
-            if (pi.getId().equals(kv[0])
-                && (kv.length != 2 || pi.getVersion().equals(kv[1]))
-                && (scope == null || gotScope.contains(scope))) {
+    }
+    if (interfaces == null) {
+      return false;
+    }
+    for (InterfaceDescriptor pi : interfaces) {
+      List<String> gotScope = pi.getScopeArray();
+      if (scope == null || gotScope.contains(scope)) {
+        String[] interfaceList = interfaceStr.split(",");
+        for (String interfacePair : interfaceList) {
+          String[] kv = interfacePair.split("=");
```

Comment:
String[][] interfacePair = new String[interfaceList.length][];
    for (int i = 0; i < interfaceList.length; i++) {
      interfacePair[i] = interfaceList[i].split("=");
    }

Splitting can be moved before the loop. `for (String[] kv : interfacePair) {`